monitoring/scd_qualifier/flight_volumes_generator.py

```diff
@@ -0,0 +1,307 @@
+        all_volume_4d :List[Volume4D] = []
+        for path_index, raw_geometry in enumerate(raw_geometries): 
+            volume_generation_rule = rules[path_index]
+            
+            flight_volume_3d = self.convert_geometry_to_volume(flight_geometry = raw_geometry.geometry, volume_generation_rule = volume_generation_rule, altitude_of_ground_level_wgs_84 = altitude_of_ground_level_wgs_84)
+            flight_volume_4d = self.transform_3d_volume_to_astm_4d(volume_3d = flight_volume_3d, volume_generation_rule = volume_generation_rule)
+            all_volume_4d.append(flight_volume_4d)
+            
+        return all_volume_4d
+
+
+class SCDFlightPathVolumeWriter():
+    """ A class to write raw Flight Paths and volumes to disk so that they can be examined / used in other software """
+
+    def __init__(self, raw_geometries:List[GeometryPayload],  flight_volumes: List[Volume4D], all_rules: List[VolumeGenerationRule],country_code='che')-> None:
+        '''
+        A method to write flight paths, volumes and volume generation rules to disk for review / submission to the test harness. All data is written in the `test_definitions` directory which is created if it does not exist.
+        
+        '''
+        self.country_code = country_code
+        self.output_directory = Path('test_definitions', self.country_code)        
+        # Create test_definition directory if it does not exist        
+        self.output_directory.mkdir(parents=True, exist_ok=True)
+        self.output_subdirectories = (Path(self.output_directory, 'astm_4d_volumes'), Path(self.output_directory, '4d_flight_geometries'),Path(self.output_directory, 'scd_rules'),)
+        for output_subdirectory in self.output_subdirectories:
+            output_subdirectory.mkdir(parents=True, exist_ok=True)
+        self.raw_geometries = raw_geometries
+        self.flight_volumes = flight_volumes
+        self.all_rules = all_rules
+
+    def write_4d_flight_geometries(self) -> None:
+        ''' A method to write flight paths to disk as GeoJSON features '''
+        
+        for path_id, raw_geometry in enumerate(self.raw_geometries):    
+            line_polygon_feature = {'type': 'Feature', 'properties': {}, 'geometry': shapely.geometry.mapping(raw_geometry.geometry)}
+            scd_volume_4d = SCDVolume4D(flight_plan= line_polygon_feature,time_start = self.flight_volumes[path_id].time_start, time_end=self.flight_volumes[path_id].time_end, altitude_lower= self.flight_volumes[path_id]['volume']['altitude_lower'], altitude_upper= self.flight_volumes[path_id]['volume']['altitude_upper'] )
+            
+            path_file_name = '4d_path_%s.json' % str(path_id + 1)  # Avoid Zero based numbering
+            tracks_file_path = self.output_subdirectories[1] / path_file_name
+            with open(tracks_file_path, 'w') as f:
+                f.write(json.dumps(scd_volume_4d))
+
+    
+    def write_astm_volume_4d(self) -> None:
```

Comment:
Thanks for this, I am re-reviewing the API definition and I think that the test driver should submit Volume 4D and the USS should generate a operational intent, I am happy to stand corrected but I don think that the test driver should submit Operational Intents.

Reply:
I have now made the changes so that the automated testing API does not accept Operational Intents but Volume 4D which was my intention, I think that the operational intents should be generated by the USS.